pkg/engine/context/context_test.go

```diff
@@ -227,3 +231,40 @@ func TestAddVariable(t *testing.T) {
 		})
 	}
 }
+
+func Test_ImageInfoLoader(t *testing.T) {
+	resource1, err := kubeutils.BytesToUnstructured([]byte(`{
+		"apiVersion": "v1",
+		"kind": "Pod",
+		"metadata": {
+		  "name": "test-pod",
+		  "namespace": "default"
+		},
+		"spec": {
+		  "containers": [{
+			"name": "test-container",
+			"image": "nginx:latest"
+		  }]
+		}
+	}`))
+	assert.Nil(t, err)
+	err = newContext().AddImageInfos(resource1, cfg)
+	assert.Nil(t, err)
+	resource2, err := kubeutils.BytesToUnstructured([]byte(`{
+		"apiVersion": "v1",
+		"kind": "Pod",
+		"metadata": {
+		  "name": "test-pod",
+		  "namespace": "default"
+		},
+		"spec": {
+		  "containers": [{
+			"name": "test-container",
+			"image": "@incorrect-image"
+		  }]
+		}
+	}`))
+	assert.Nil(t, err)
+	err = newContext().AddImageInfos(resource2, cfg)
+	assert.Nil(t, err)
+}
```

Comment:
What is this test doing? 

We need to check if the deferred loader was triggered or not.

The expectation is that unless there is an `images` variable the ImageInfo data is not resolved.

But what happens if `context.ImageInfo` is programmatically called?

Seems like we want 3 tests:
1. Build context for a policy with no `images` variable, and verify the ImageInfo data is not loaded.
2. Build context for a policy with an `images` variable, and verify the ImageInfo data is loaded.
3. Build context for a policy with no `images` variable, and call `context.Images()` and make sure it returns the right data.

Reply:
I have edited the test to achieve the same. We see that when we `AddImageInfo` it is nil initially till we force call/query ImageInfo/"images".

P.S. Initially, I had tried setting context with a policy and resource, it didn't work out without mocking the context (adding other data to the JSON context).

This ensures the desired outcome, WDYT?